pyhive/presto.py

```diff
@@ -172,6 +172,16 @@ def execute(self, operation, parameters=None):
         response = requests.post(url, data=sql.encode('utf-8'), headers=headers)
         self._process_response(response)
 
+    def cancel(self):
+        if self._state == self._STATE_NONE:
+            raise ProgrammingError("No query yet")
+        if self._nextUri is None:
+            assert self._state == self._STATE_FINISHED, "Should be finished if nextUri is None"
+            return None
+        response = requests.delete(self._nextUri)
+        self._process_response(response)
+        return response.status_code
```

Comment:
Sounds good to me. I wasn't completely sure what to return, figured the status code would give some indication of "What you expected to happen, happened".